packages/aws-cdk/lib/api/hotswap/cloudformation-executable-template.ts

```diff
@@ -0,0 +1,230 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import * as AWS from 'aws-sdk';
+import { ListStackResources } from './common';
+
+export interface CloudFormationExecutableTemplateProps {
+  readonly stackArtifact: cxapi.CloudFormationStackArtifact;
+  readonly parameters: { [parameterName: string]: string };
+  readonly account: string;
+  readonly region: string;
+  readonly partition: string;
+  readonly urlSuffix: string;
+
+  readonly listStackResources: ListStackResources;
+}
+
+export class CloudFormationExecutableTemplate {
+  private readonly stackResources: ListStackResources;
+  private readonly context: { [k: string]: string };
+  private readonly account: string;
+  private readonly region: string;
+  private readonly partition: string;
+
+  constructor(props: CloudFormationExecutableTemplateProps) {
+    this.stackResources = props.listStackResources;
+    this.context = {
+      'AWS::AccountId': props.account,
+      'AWS::Region': props.region,
+      'AWS::Partition': props.partition,
+      'AWS::URLSuffix': props.urlSuffix,
+      ...props.parameters,
+    };
+    this.account = props.account;
+    this.region = props.region;
+    this.partition = props.partition;
+  }
+
+  public async findPhysicalNameFor(logicalId: string): Promise<string | undefined> {
+    const stackResources = await this.stackResources.listStackResources();
+    return stackResources.find(sr => sr.LogicalResourceId === logicalId)?.PhysicalResourceId;
+  }
+
+  public async evaluateCfnExpression(cfnExpression: any): Promise<any> {
+    if (cfnExpression == null) {
+      return cfnExpression;
+    }
+
+    if (Array.isArray(cfnExpression)) {
+      return Promise.all(cfnExpression.map(expr => this.evaluateCfnExpression(expr)));
+    }
+
+    if (typeof cfnExpression === 'object') {
+      const intrinsic = this.parseIntrinsic(cfnExpression);
+      if (intrinsic) {
+        return this.evaluateIntrinsic(intrinsic);
+      } else {
+        const ret: { [key: string]: any } = {};
+        for (const [key, val] of Object.entries(cfnExpression)) {
+          ret[key] = await this.evaluateCfnExpression(val);
+        }
+        return ret;
+      }
+    }
+
+    return cfnExpression;
+  }
+
+  async 'Fn::Join'(separator: string, args: any[]): Promise<string> {
```

Comment:
💅  I'm not super down with these dispatch handlers just living among all the other normal methods on this class. I'd prefer for them to be in a different class, or object, or *something* to distinguish them from the rest of the code here.

Reply:
Yep. I didn't like it either, but laziness got the better of me. Fixed it now.